server-base/src/main/java/org/apache/kylin/rest/service/ProjectService.java

```diff
@@ -137,8 +138,10 @@ public ProjectInstance updateProject(ProjectInstance newProject, ProjectInstance
     @PreAuthorize(Constant.ACCESS_HAS_ROLE_ADMIN)
     public void deleteProject(String projectName, ProjectInstance project) throws IOException {
         Set<String> tables = project.getTables();
-        for (String table : tables) {
+        for (String table : Sets.newTreeSet(tables)) {
```

Comment:
What's the reason to create a new set? Can you provide more information?

Reply:
Since the unloading table operation will modify the set tables,while tables is traversed at the same time,which will cause concurrentException.

Reply:
I see, thank you!